slides/week6/week6.org

```diff
@@ -0,0 +1,57 @@
+#+TITLE: Week 6
+#+AUTHOR: Sahit Chintalapudi 
+#+EMAIL: schintalapudi@gatech.edu
+
+* Computer Vision
+- We have the knowledge in C++ to describe the logic we might want a robot to
+  have. But we need to be able to make sense of what the robot sees and
+  classify it before we can act on this logic.
+- Cue OpenCV, an open source computer vision library with bindings for C++
+  (and a few other languages)
+- I guess our ability to see has been ++'d
+
+* What does an Image look like to your computer?
+- OpenCV stores images in an object called a /Mat/
+- A Mat is matrix with rows and columns! In a grayscale image. each element
+  of the Mat is a pixel in the image and its location in the Mat corresponds 
+  to its location in the image
+- If the image is grayscale, then a pixel is an int from 0 (black) to 255
+  (white)
+- Computers have no concept of "2d", so Images in memory are /continuous/.
+  This means each row of the image is appended onto the end of the last. To
+  iterate through a Mat you just get a pointer to the beginning of the first
+  row and keep track of your row number by how far you've traversed.
+
+* Color Images
+- Color images don't embed the color of a pixel in one element. Often, you'll
+  find each pixel represented in BGR (Blue component, Green Component, Red
+  Component) form. So now, each row of a color image is 3 times as long as a
+  row of a black and white image.
+[[https://docs.opencv.org/2.4/_images/math/b6df115410caafea291ceb011f19cc4a19ae6c2c.png]]
+- Images don't have to be stored in just BGR format!
+
+* HSV Images
+- Each Pixel in a color image has a hue, a saturation, and a luminosity. 
+- The nice thing about looking at colors from this perspective is that we're
+  more tolerant to changes in lighting conditions making the same color look
+  different.
+[[https://image.slidesharecdn.com/01presentationhuehistograms-150707215651-lva1-app6892/95/about-perception-and-hue-histograms-in-hsv-space-5-638.jpg]]
+
+* Finding the blue in an image
+#+BEGIN_SRC c++
+Mat findBlue(const Mat& frameBGR) { 
```

Comment:
so what i mean by 

> The cvt in cvtColor is convert because ....

is that "cvtColor" is literally shorthand for "convertColor". Not sure if I said that clearly before. If I did, and you were asking for source to prove that the word cvt was short for the word convert than the documentation on the method might be all that I can do for you and that isn't very explicit. 
http://docs.opencv.org/2.4/modules/imgproc/doc/miscellaneous_transformations.html#cvtcolor

As for why the low HSV seems green, I see the issue. It was in our best interests to err on the wide side for our color bounds for blue because not a lot else on the track was blue or neighboring colors (that green that you see). Any noisy pixels would be filtered out by later postprocessing. Those bounds were the actual ones we were going to use for IARRC this year where only the finish line was blue.